diff --git a/vendor/lowrisc_ip/ip/prim_generic/rtl/prim_generic_clock_gating.sv b/vendor/lowrisc_ip/ip/prim_generic/rtl/prim_generic_clock_gating.sv
index 1cc05e6..e939112 100644
--- a/vendor/lowrisc_ip/ip/prim_generic/rtl/prim_generic_clock_gating.sv
+++ b/vendor/lowrisc_ip/ip/prim_generic/rtl/prim_generic_clock_gating.sv
@@ -13,13 +13,6 @@ module prim_generic_clock_gating #(
   output logic clk_o
 );
 
-  // Assume en_i synchronized, if not put synchronizer prior to en_i
-  logic en_latch;
-  always_latch begin
-    if (!clk_i) begin
-      en_latch = en_i | test_en_i;
-    end
-  end
-  assign clk_o = en_latch & clk_i;
+  assign clk_o = clk_i;
 
 endmodule
