diff --git a/tests/regression.tcl b/tests/regression.tcl
index 0fa6c2a55..24464d0a6 100755
--- a/tests/regression.tcl
+++ b/tests/regression.tcl
@@ -79,7 +79,7 @@ proc log_nonewline { text } {
 }
 
 set UPDATE 0
-set TIME "/usr/bin/env time"
+set TIME "/usr/bin/time"
 set DEBUG_TOOL ""
 set PRIOR_USER 0
 set PRIOR_ELAPSED 0
@@ -209,7 +209,7 @@ set UHDM_DUMP_COMMAND "[pwd]/third_party/UHDM/bin/uhdm-dump"
 
 set REGRESSION_PATH [pwd]
 
-set SURELOG_COMMAND "$TIME $DEBUG_TOOL $SURELOG_VERSION"
+set SURELOG_COMMAND "$TIME $DEBUG_TOOL $EXE_NAME"
 
 source [project_path]/tests/blacklisted.tcl
 
@@ -453,7 +453,7 @@ proc formal_verification { command testname } {
     close $yid
     set surelog_parse ""
     catch {set out [exec $YOSYS_EXE -s "$output_dir/surelog.ys" -q -q -l $output_dir/surelog.out]} surelog_parse
-    
+
     # Yosys parser
     set yid [open "$output_dir/yosys.ys" "w"]
     puts $yid "read_verilog -sv $yosys_command"
@@ -483,11 +483,10 @@ proc formal_verification { command testname } {
         close $fid
     }
     # Find cell libraries
-    set yosys_path "$env(HOME)/yosys"
-    set cells_sim $yosys_path/share/xilinx/cells_sim.v
-    set cells_xtra $yosys_path/share/xilinx/cells_xtra.v
-    if [file exist [file dirname $YOSYS_EXE]/../share/yosys/xilinx/cells_sim.v] {
-        set yosys_path [file dirname $YOSYS_EXE]/..
+    set yosys_path ""
+    catch {exec which $YOSYS_EXE} yosys_path
+    if [file exist [file dirname $yosys_path]/../share/yosys/xilinx/cells_sim.v] {
+        set yosys_path [file dirname $yosys_path]/..
         set cells_sim $yosys_path/share/yosys/xilinx/cells_sim.v
         set cells_xtra $yosys_path/share/yosys/xilinx/cells_xtra.v
     }
@@ -820,8 +819,8 @@ proc run_regression { } {
             close $fid
             set no_previous_time_content 0
         }
-        if [file exists "$testname.log"] {
-            set fid [open "$testname.log" "r"]
+        if [file exists "$REGRESSION_PATH/tests/$test/$testname.log"] {
+            set fid [open "$REGRESSION_PATH/tests/$test/$testname.log" "r"]
             set content [read $fid]
             close $fid
             foreach {log_fatals log_errors log_warnings log_notes log_details log_syntax} [count_messages $content] {}
