diff --git a/tests/regression.tcl b/tests/regression.tcl
index 0fa6c2a55..8d886a9de 100755
--- a/tests/regression.tcl
+++ b/tests/regression.tcl
@@ -209,7 +209,7 @@ set UHDM_DUMP_COMMAND "[pwd]/third_party/UHDM/bin/uhdm-dump"
 
 set REGRESSION_PATH [pwd]
 
-set SURELOG_COMMAND "$TIME $DEBUG_TOOL $SURELOG_VERSION"
+set SURELOG_COMMAND "surelog"
 
 source [project_path]/tests/blacklisted.tcl
 
@@ -820,8 +820,8 @@ proc run_regression { } {
             close $fid
             set no_previous_time_content 0
         }
-        if [file exists "$testname.log"] {
-            set fid [open "$testname.log" "r"]
+        if [file exists "$REGRESSION_PATH/tests/$test/$testname.log"] {
+            set fid [open "$REGRESSION_PATH/tests/$test/$testname.log" "r"]
             set content [read $fid]
             close $fid
             foreach {log_fatals log_errors log_warnings log_notes log_details log_syntax} [count_messages $content] {}
