diff --git a/bp_top/syn/flist.vcs b/bp_top/syn/flist.vcs
index 7aec15da..5f9e5035 100644
--- a/bp_top/syn/flist.vcs
+++ b/bp_top/syn/flist.vcs
@@ -82,15 +82,13 @@ $BASEJUMP_STL_DIR/bsg_mem/bsg_cam_1r1w_sync.sv
 $BASEJUMP_STL_DIR/bsg_mem/bsg_cam_1r1w_tag_array.sv
 $BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1r1w.sv
 $BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1r1w_one_hot.sv
-$BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1r1w_sync_synth.sv
-$BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1r1w_sync.sv
 $BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1r1w_synth.sv
-$BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1rw_sync.sv
-$BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1rw_sync_mask_write_bit.sv
-$BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1rw_sync_mask_write_bit_synth.sv
-$BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1rw_sync_mask_write_byte.sv
-$BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1rw_sync_mask_write_byte_synth.sv
-$BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1rw_sync_synth.sv
+$BASEJUMP_STL_DIR/bsg_mem/bsg_mem_1rw_sync_mask_write_bit_from_1r1w.sv
+$BASEJUMP_STL_DIR/hard/ultrascale_plus/bsg_mem/bsg_mem_1r1w_sync.sv
+$BASEJUMP_STL_DIR/hard/ultrascale_plus/bsg_mem/bsg_mem_1r1w_sync_mask_write_byte.sv
+$BASEJUMP_STL_DIR/hard/ultrascale_plus/bsg_mem/bsg_mem_1rw_sync.sv
+$BASEJUMP_STL_DIR/hard/ultrascale_plus/bsg_mem/bsg_mem_1rw_sync_mask_write_bit.sv
+$BASEJUMP_STL_DIR/hard/ultrascale_plus/bsg_mem/bsg_mem_1rw_sync_mask_write_byte.sv
 $BASEJUMP_STL_DIR/bsg_mem/bsg_mem_2r1w_sync.sv
 $BASEJUMP_STL_DIR/bsg_mem/bsg_mem_2r1w_sync_synth.sv
 $BASEJUMP_STL_DIR/bsg_mem/bsg_mem_3r1w_sync.sv
diff --git a/external/basejump_stl b/external/basejump_stl
--- a/external/basejump_stl
+++ b/external/basejump_stl
@@ -1 +1 @@
-Subproject commit de84d1100f7c786e8006e26bb806f804cf6a3ce8
+Subproject commit de84d1100f7c786e8006e26bb806f804cf6a3ce8-dirty
