From 9b0e8a2bf8963357966da8fc8e06472777ac7484 Mon Sep 17 00:00:00 2001
From: Kamil Rakoczy <krakoczy@antmicro.com>
Date: Thu, 20 Apr 2023 17:26:12 +0200
Subject: [PATCH] Replace stream-op

Signed-off-by: Kamil Rakoczy <krakoczy@antmicro.com>
---
 bsg_misc/bsg_scan.v | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/bsg_misc/bsg_scan.v b/bsg_misc/bsg_scan.v
index 063e74ef..c5433edb 100644
--- a/bsg_misc/bsg_scan.v
+++ b/bsg_misc/bsg_scan.v
@@ -66,7 +66,9 @@ module bsg_scan #(parameter `BSG_INV_PARAM(width_p)
 
    // streaming operation; reverses bits
    if (lo_to_hi_p)
-     assign t[0] = {<< {i}};
+     for (genvar i = width_p-1; i >= 0; i--) begin
+       assign t[width_p-1-i] = t[i];
+     end
    else
      assign t[0] = i;
 
-- 
2.39.0

