diff --git a/hw/ip/otbn/rtl/otbn_decoder.sv b/hw/ip/otbn/rtl/otbn_decoder.sv
index 8cd4814d7..90fdd2b17 100644
--- a/hw/ip/otbn/rtl/otbn_decoder.sv
+++ b/hw/ip/otbn/rtl/otbn_decoder.sv
@@ -123,7 +123,7 @@ module otbn_decoder
   // Decoder //
   /////////////
 
-  always_comb begin
+  always begin
     rf_wdata_sel          = RfWdSelEx;
     rf_we                 = 1'b0;
     rf_ren_a              = 1'b0;
diff --git a/hw/ip/padctrl/rtl/padring.sv b/hw/ip/padctrl/rtl/padring.sv
index fc8b3d183..6a5b97985 100644
--- a/hw/ip/padctrl/rtl/padring.sv
+++ b/hw/ip/padctrl/rtl/padring.sv
@@ -151,7 +151,7 @@ module padring import padctrl_reg_pkg::*; #(
     end else begin : gen_mio_tie_off
       logic unused_out, unused_oe, unused_pad;
       logic [AttrDw-1:0] unused_attr;
-      assign mio_pad_io[k] = 1'b0;
+      //assign mio_pad_io[k] = 1'b0;
       assign unused_pad   = mio_pad_io[k];
       assign unused_out   = mio_out_i[k];
       assign unused_oe    = mio_oe_i[k];
@@ -213,7 +213,7 @@ module padring import padctrl_reg_pkg::*; #(
     end else begin : gen_dio_tie_off
       logic unused_out, unused_oe, unused_pad;
       logic [AttrDw-1:0] unused_attr;
-      assign dio_pad_io[k] = 1'b0;
+      //assign dio_pad_io[k] = 1'b0;
       assign unused_pad   = dio_pad_io[k];
       assign unused_out   = dio_out_i[k];
       assign unused_oe    = dio_oe_i[k];
diff --git a/hw/ip/prim/util/primgen.py b/hw/ip/prim/util/primgen.py
index f601d503e..70dd6d438 100755
--- a/hw/ip/prim/util/primgen.py
+++ b/hw/ip/prim/util/primgen.py
@@ -238,6 +238,22 @@ def _generate_prim_pkg(gapi):
     shutil.copyfile(prim_pkg_core_src, prim_pkg_core_dest)
     print("Core file written to %s." % (prim_pkg_core_dest, ))
 
+def _instance_sv_ram(prim_name, techlib, parameters, prim_type):
+    if not parameters:
+        s = "    prim_{techlib}_{prim_name} u_impl_{techlib} (\n"
+    else:
+        s = "    prim_{techlib}_{prim_name} #(\n"
+        s += ",\n".join("      .{p}({p})".format(p=p) for p in parameters)
+        s += "\n    ) u_impl_{techlib}_{prim_type} (\n"
+    s += "      .clk_i(clk_{prim_type}_i),\n"
+    s += "      .req_i({prim_type}_req_i),\n"
+    s += "      .write_i({prim_type}_write_i),\n"
+    s += "      .addr_i({prim_type}_addr_i),\n"
+    s += "      .wdata_i({prim_type}_wdata_i),\n"
+    s += "      .wmask_i({prim_type}_wmask_i),\n"
+    s += "      .rdata_o({prim_type}_rdata_o)\n"
+    s +=  "    );\n"
+    return s.format(prim_name=prim_name, techlib=techlib, prim_type=prim_type)
 
 def _instance_sv(prim_name, techlib, parameters):
     if not parameters:
@@ -246,8 +261,8 @@ def _instance_sv(prim_name, techlib, parameters):
         s = "    prim_{techlib}_{prim_name} #(\n"
         s += ",\n".join("      .{p}({p})".format(p=p) for p in parameters)
         s += "\n    ) u_impl_{techlib} (\n"
-    s += "      .*\n" \
-         "    );\n"
+    s += "      .*\n"
+    s +=  "    );\n"
     return s.format(prim_name=prim_name, techlib=techlib)
 
 
@@ -258,12 +273,17 @@ def _create_instances(prim_name, techlibs, parameters):
     ]
     techlibs_generic_last = techlibs_wo_generic + ['generic']
 
+
     if not techlibs_wo_generic:
         # Don't output the if/else blocks if there no alternatives exist.
         # We still want the generate block to keep hierarchical path names
         # stable, even if more than one techlib is found.
         s = "  if (1) begin : gen_generic\n"
-        s += _instance_sv(prim_name, "generic", parameters) + "\n"
+        if prim_name == "ram_2p":
+            s += _instance_sv_ram("ram_1p", "generic", parameters, "a") + "\n"
+            s += _instance_sv_ram("ram_1p", "generic", parameters, "b") + "\n"
+        else:
+            s += _instance_sv(prim_name, "generic", parameters) + "\n"
         s += "  end"
         return s
 
@@ -282,8 +302,14 @@ def _create_instances(prim_name, techlibs, parameters):
         # TODO: wildcard port lists are against our style guide, but it's safer
         # to let the synthesis tool figure out the connectivity than us trying
         # to parse the port list into individual signals.
-        s += "begin : gen_{techlib}\n" + _instance_sv(prim_name, techlib,
-                                                      parameters) + "end"
+        s += "begin : gen_{techlib}\n" 
+        if prim_name == "ram_2p":
+            s += _instance_sv_ram("ram_1p", techlib, parameters, "a")
+            s += _instance_sv_ram("ram_1p", techlib, parameters, "b")
+        else:
+            s += _instance_sv(prim_name, techlib, parameters)
+
+        s += "end"
 
         if not is_last:
             s += " "
diff --git a/hw/ip/rstmgr/rtl/rstmgr_pkg.sv b/hw/ip/rstmgr/rtl/rstmgr_pkg.sv
index 9d4ea8b6a..f8478dbe0 100644
--- a/hw/ip/rstmgr/rtl/rstmgr_pkg.sv
+++ b/hw/ip/rstmgr/rtl/rstmgr_pkg.sv
@@ -38,6 +38,7 @@ package rstmgr_pkg;
     logic rst_por_n;
     logic rst_por_io_n;
     logic rst_por_io_div2_n;
+    logic rst_por_io_div4_n;
     logic rst_por_usb_n;
     logic rst_lc_n;
     logic rst_sys_n;
@@ -53,6 +54,13 @@ package rstmgr_pkg;
     logic ndmreset_req;
   } rstmgr_cpu_t;

+  // exported resets
+  typedef struct packed {
+    logic rst_ast_usbdev_sys_io_n;
+    logic rst_ast_usbdev_usb_n;
+    logic rst_ast_sensor_ctrl_sys_io_n;
+  } rstmgr_ast_out_t;
+
   // default value for rstmgr_ast_rsp_t (for dangling ports)
   parameter rstmgr_cpu_t RSTMGR_CPU_DEFAULT = '{
     rst_cpu_n: 1'b1,
diff --git a/hw/top_earlgrey/top_earlgrey_nexysvideo.core b/hw/top_earlgrey/top_earlgrey_nexysvideo.core
index 8d6cf89b6..6480e08ba 100644
--- a/hw/top_earlgrey/top_earlgrey_nexysvideo.core
+++ b/hw/top_earlgrey/top_earlgrey_nexysvideo.core
@@ -56,7 +56,7 @@ targets:
     filesets:
       - files_rtl_nexysvideo
       - files_constraints
-      - files_tcl
+      #- files_tcl
     toplevel: top_earlgrey_nexysvideo
     parameters:
       - BootRomInitFile
@@ -64,6 +64,16 @@ targets:
     tools:
       vivado:
         part: "xc7a200tsbg484-1" # Nexys Video
+        synth: "yosys"
+        yosys_synth_options: ['-flatten', '-iopad', '-noclkbuf', '-family xc7', "frontend=surelog"]
+        yosys_read_options: ['-noassert', '-dump_ast1', '-dump_ast2', '-no_dump_ptr']
+        yosys_extra_passes: ['delete */t:\$scopeinfo']
+        surelog_options: ['-DSYNTHESIS', '-synth']
+      yosys:
+        arch: "xilinx"
+        yosys_synth_options: ['-flatten', '-iopad', '-noclkbuf', '-family xc7', "frontend=surelog"]
+        yosys_read_options: ['-noassert', '-dump_ast1', '-dump_ast2', '-no_dump_ptr']
+        surelog_options: ['-DSYNTHESIS', '-synth']
 
   lint:
     <<: *default_target
