diff --git a/hw/top_custom_tiny/chip_custom_tiny_nexysvideo.core b/hw/top_custom_tiny/chip_custom_tiny_nexysvideo.core
index 47ef3473b..3f2da9c9e 100644
--- a/hw/top_custom_tiny/chip_custom_tiny_nexysvideo.core
+++ b/hw/top_custom_tiny/chip_custom_tiny_nexysvideo.core
@@ -83,7 +83,7 @@ targets:
       - AST_BYPASS_CLK=true
 
   synth:
-    default_tool: vivado
+    default_tool: yosys
     filesets:
       - files_rtl_nexysvideo
       - files_constraints
@@ -96,8 +96,9 @@ targets:
       - IBEX_CUSTOM_PMP_RESET_VALUES
       - AST_BYPASS_CLK=true
     tools:
-      vivado:
-        part: "xc7a200tsbg484-1"
+      yosys:
+        arch: "xilinx"
+        # part: "xc7a200tsbg484-1"
 
   lint:
     <<: *default_target
